core-java-modules/core-java-8-2/src/test/java/com/baeldung/javaClass/ClassForNameAndInstanceUnitTest.java

```diff
@@ -0,0 +1,19 @@
+package com.baeldung.javaClass;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Test;
+
+public class ClassForNameAndInstanceUnitTest {
+    ClassForNameAndInstance tester= new ClassForNameAndInstance();
+    @Test
+    public void givenString_whenValidClass_thenReturnClass() throws ClassNotFoundException {
+        Class c= tester.createClassForName("java.lang.String");
+        assertEquals(c.getName(),"java.lang.String");
+    }
+    @Test
+    public void givenString_whenValidClass_thenReturnClassObject() throws ClassNotFoundException, InstantiationException, IllegalAccessException {
+        Object o= tester.createNewInstance("java.lang.String");
```

Comment:
Creating a new instance of `String` isn't very interesting. What if, instead, we created a custom class here in the example code and tried to use its constructor. We could even give it some default field values so we can assert that it got created properly.